temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowOutboundCallsInterceptor.java

```diff
@@ -0,0 +1,104 @@
+import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptorBase;
+import io.temporal.opentracing.OpenTracingConfig;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowInfo;
+
+public class OpenTracingWorkflowOutboundCallsInterceptor
+    extends WorkflowOutboundCallsInterceptorBase {
+  private final OpenTracingConfig openTracingConfig = OpenTracingConfig.newBuilder().build();
+
+  public OpenTracingWorkflowOutboundCallsInterceptor(WorkflowOutboundCallsInterceptor next) {
+    super(next);
+  }
+
+  @Override
+  public <R> ActivityOutput<R> executeActivity(ActivityInput<R> input) {
+    createAndPassActivityStartSpan(input.getActivityName(), input.getHeader());
+    return super.executeActivity(input);
+  }
+
+  @Override
+  public <R> LocalActivityOutput<R> executeLocalActivity(LocalActivityInput<R> input) {
+    createAndPassActivityStartSpan(input.getActivityName(), input.getHeader());
+    return super.executeLocalActivity(input);
+  }
+
+  @Override
+  public <R> ChildWorkflowOutput<R> executeChildWorkflow(ChildWorkflowInput<R> input) {
+    createAndPassChildWorkflowStartSpan(input);
+    return super.executeChildWorkflow(input);
+  }
+
+  private void createAndPassActivityStartSpan(String activityName, Header header) {
+    if (Workflow.isReplaying()) {
+      return;
+    }
+    Tracer tracer = GlobalTracer.get();
+    Span activitySpan = createActivityStartSpanBuilder(tracer, activityName).start();
+    activitySpan.finish();
+    OpenTracingContextPropagationUtils.writeSpanContextToHeader(
+        activitySpan.context(), header, tracer);
+  }
+
+  private Tracer.SpanBuilder createActivityStartSpanBuilder(Tracer tracer, String activityName) {
+    WorkflowInfo workflowInfo = Workflow.getInfo();
+    return OpenTracingSpanUtils.createActivityStartSpan(
+        tracer,
+        Workflow.currentTimeMillis(),
+        openTracingConfig.getActivityStartOperationNamePrefix() + "-" + activityName,
+        workflowInfo.getWorkflowId(),
+        workflowInfo.getRunId());
+  }
+
+  private <R> void createAndPassChildWorkflowStartSpan(ChildWorkflowInput<R> input) {
+    if (Workflow.isReplaying()) {
+      return;
+    }
+    Tracer tracer = GlobalTracer.get();
+    Tracer.SpanBuilder workflowSpan = createChildWorkflowStartSpanBuilder(tracer, input);
+    Span span = workflowSpan.start();
+    span.finish();
+    OpenTracingContextPropagationUtils.writeSpanContextToHeader(
+        span.context(), input.getHeader(), tracer);
+  }
+
+  private <R> Tracer.SpanBuilder createChildWorkflowStartSpanBuilder(
+      Tracer tracer, ChildWorkflowInput<R> input) {
+    WorkflowInfo parentWorkflowInfo = Workflow.getInfo();
+    return OpenTracingSpanUtils.createChildWorkflowStartSpan(
+        tracer,
+        System.currentTimeMillis(),
+        openTracingConfig.getChildWorkflowStartOperationNamePrefix()
+            + "-"
+            + input.getWorkflowType(),
```

Comment:
Same as above, should we put workflow type into tags instead of part of the operation name?